docs/book/src/developer/providers/contracts/control-plane.md

```diff
@@ -59,196 +318,502 @@ status: {}
+
+If a condition with type `Ready` exist, such condition will be mirrored in Cluster's `ControlPlaneReady` condition.
+
+Please note that the `Ready` condition is expected to surface the status of the ControlPlane during its own entire lifecycle,
+including initial provisioning, the final deletion process, and the period in between these two moments.
+
+See [Cluster API condition proposal] for more context.
+
+<aside class="note warning">
+
+<h1>Heads up! this will change with the v1beta2 contract</h1>
+
+When the v1beta2 contract will be released (tentative Apr 2025), Cluster API will start using Kubernetes metav1.Condition
+types and fully comply to [Kubernetes API Conventions].
+
+In order to support providers continuing to use legacy Cluster API condition types, providers transitioning to
+metav1.Condition or even providers adopting custom condition types, Cluster API will start to accept conditions that
+provides following information:
+- `type`
+- `status`
+- `reason` ((optional, if omitted, a default one will be used)
+- `message` (optional)
+- `lastTransitionTime` (optional, if omitted, time.Now will be used)
+
+Other fields will be ignored.
+
+Additional considerations apply specifically to the ControlPlane resource:
+
+In order to disambiguate the usage of the ready term and improve how the status of the control plane is
+presented, Cluster API will stop surfacing the `Ready` condition and instead surface a new `Available` condition.
+
+The the `Available` condition is expected to properly represents the fact that a ControlPlane can be operational 
+even if there is a certain degree of not readiness / disruption in the system, or if lifecycle operations are happening.
+
+Last, but not least, in order to ensure a consistent users experience, it is also recommended to consider aligning also other 
+ControlPlane conditions to conditions existing on other Cluster API objects.  
+
+For example `KubeadmControlPlane` is implements following conditions on top of the `Available` defined by this contract:
+`CertificatesAvailable`, `EtcdClusterAvailable`, `MachinesReady`, `MachinesUpToDate`, `ScalingUp`, `ScalingDown`, `Remediating`, `Deleting`, `Paused`.
+
+See [Improving status in CAPI resources] for more context.
+
+Please also note that provider that will continue to use legacy Cluster API condition types MUST carefully take into account
+the implication of this choice which are described both in the document above and in the notice at the beginning of the [Cluster API condition proposal]..
+
+</aside>
+
+### ControlPlane: terminal failures
+
+Each ControlPlane SHOULD report when Cluster's enter in a state that cannot be recovered (terminal failure) by
+setting `status.failureReason` and `status.failureMessage` in the ControlPlane resource.
+
+```go
+type FoControlPlaneStatus struct {
+    // FailureReason will be set in the event that there is a terminal problem reconciling the FooControlPlane
```

Comment:
nit: given other changes in this PR you probably want to use failureReason (same for failureMessage below)